src/wallet/walletdb.h

```diff
@@ -124,9 +125,11 @@ class CKeyMetadata
         std::vector<std::string> nComponents;
         if(hdKeypath=="m")
             return false;
-        boost::split(nComponents, hdKeypath, boost::is_any_of("/"), boost::token_compress_on);
-        nChange = boost::lexical_cast<int64_t>(nComponents[4]);
-        nChild = boost::lexical_cast<int64_t>(nComponents[5]);
+        std::string hdKeypathNonHardened = hdKeypath;
+        boost::erase_all(hdKeypathNonHardened, "'");
+        boost::split(nComponents, hdKeypathNonHardened, boost::is_any_of("/"), boost::token_compress_on);
+        nChange = boost::lexical_cast<int64_t>(nComponents[nComponents.size()-2]);
+        nChild = boost::lexical_cast<int64_t>(nComponents[nComponents.size()-1]);
```

Comment:
I think we should support or return false when we found hardened key instead of treat it as non-hardened.

Reply:
Yes this is good suggestion, and I have actually extended this code to include a "Keypath" class, which will be needed when we add BIP39 support. However for now, `nChange` and `nChild` are only used by the BIP44 path, so this works without a big change.

Reply:
Then we should return false in hardened case. To let caller handle it for this release.

Reply:
I don't understand what you mean. Do you mean adding `isHardened` books for `nChange` and `nChild`?

Reply:
@thebevrishot Added information related to "hardened" values.